configure.ac

```diff
@@ -36,52 +49,162 @@ AC_PROG_MKDIR_P
+                if test "$enableval" = "yes"; then
+                    build_tools=yes
+                    build_services=yes
+                    build_apis=yes
+                fi
+              ],
+              [])
+
+AC_ARG_ENABLE([build-services],
+              [AS_HELP_STRING([--enable-build-services], [Enable services build (enabled by default)])],
+              [
+                if test "$enableval" = "yes"; then
+                    build_services=yes
+                    build_apis=yes
+                else
+                    build_services=no
+                    build_tools=no
+                fi
+              ],
+              [])
+
+AC_ARG_ENABLE([build-apis],
+              [AS_HELP_STRING([--enable-build-apis], [Enable APIs build (enabled by default)])],
+              [
+                if test "$enableval" = "yes"; then
+                    build_apis=yes
+                else
+                    build_apis=no
+                    build_services=no
+                    build_tools=no
+                fi
+              ],
+              [])
+
+AC_ARG_ENABLE([build-inscribeit],
+              [AS_HELP_STRING([--enable-build-inscribeit], [Enable inscribeit API build (disabled by default)])],
+              [build_inscribeit=yes]
+              [])
+
+
+if test "$plugin_api" = "yes"; then
+    if test "$build_apis" = "yes"; then
+        AC_MSG_ERROR([Cannot enable plugin API with general APIs])
     fi
+fi
 
-    AC_MSG_NOTICE([python is $PYTHON])
-
-    PYBIND_CPPFLAGS=`$PYTHON -m pybind11 --includes`
-
-    AC_MSG_NOTICE([PYBIND_CPPFLAGS = $PYBIND_CPPFLAGS])
-
-    ax_check_pybind_cppflags=$CPPFLAGS
+AM_CONDITIONAL([BUILD_APIS], [test "$build_apis" = "yes"])
+AM_CONDITIONAL([BUILD_SERVICES], [test "$build_services" = "yes"])
+AM_CONDITIONAL([BUILD_TOOLS], [test "$build_tools" = "yes"])
+AM_CONDITIONAL([BUILD_INSCRIBEIT], [test "$build_inscribeit" = "yes"])
+AM_CONDITIONAL([BUILD_PLUGIN_API], [test "$plugin_api" = "yes"])
+AM_CONDITIONAL([BUILD_PYTHON_MODULE], [test "$python_module" = "yes"])
+AM_CONDITIONAL([BUILD_WASM_MODULE], [test "$wasm_module" = "yes"])
 
-    AC_LANG_SAVE
-    AC_LANG([C++])
+if test "$python_module" = "yes"; then
+    AM_PATH_PYTHON([3])
+    AC_MSG_NOTICE([python is $PYTHON])
 
-    CPPFLAGS="$CPPFLAGS $PYBIND_CPPFLAGS"
-    AC_CHECK_HEADER([pybind11/pybind11.h], [], [AC_MSG_ERROR([PyBind11 C++ header are required but not found])])
-    CPPFLAGS=$ax_check_pybind_cppflags
+    AX_PKG_SWIG([], [], [ AC_MSG_ERROR([SWIG is required to build.]) ])
+    AX_PYTHON_DEVEL
+    AX_SWIG_PYTHON
 
-    AC_LANG_RESTORE
+    AC_CHECK_PROGS([PYTHON3_CONFIG], [python3-config python-config])
+    if test -n "$PYTHON3_CONFIG"; then
+        PYTHON_LIBS=`$PYTHON3_CONFIG --libs`
+        PYTHON_LDFLAGS=`$PYTHON3_CONFIG --ldflags`
+    else
+        AC_MSG_ERROR([python3-config not found. Please install the python3-dev or python3-devel package.])
+    fi
+    AC_SUBST(PYTHON_LIBS)
+    AC_SUBST(PYTHON_LDFLAGS)
```

Comment:
Do we need these lines since AX_PYTHON_DEVEL. According to AX_PYTHON_DEVEL docs:
This macro checks for Python and tries to get the include path to ’Python.h’. It provides the \\$(PYTHON_CPPFLAGS) and $(PYTHON_LIBS) output variables. It also exports \\$(PYTHON_EXTRA_LIBS) and \\$(PYTHON_EXTRA_LDFLAGS) for embedding Python in your code.